client/src/components/Navbar.jsx

```diff
@@ -282,12 +282,26 @@ const DesktopSubNav = ({ label, href, subLabel }) => {
           ? () => {}
           : () =>
               modalComponentStore.openModal(
-                modalTitle,
-                <ModalBody />,
-                <ModalFooter />,
+                matchingModalTitle,
+                <MatchingModalBody />,
+                <MatchingModalFooter />,
                 () => {
```

Comment:
can help me add `e` as an argument for this? then first line in the function call `e.preventDefault()`. hahahaha i completely forgot about this, but forms `onSubmit` will call `react-refresh`. that's why the page keeps on refreshing haha, but not sure why we can avoid it everywhere else though. maybe because all our other submit functions use Chakra's useToast hook which somehow settles it for us.